pkg/operator/mon/health.go

```diff
@@ -100,27 +135,41 @@ func (c *Cluster) checkHealth() error {
 				continue
 			}
 
-			if len(status.MonMap.Mons) > c.Size {
-				// no need to create a new mon since we have an extra
-				err = c.removeMon(mon.Name)
-				if err != nil {
-					logger.Errorf("failed to remove mon %s. %+v", mon.Name, err)
-				}
-			} else {
-				// bring up a new mon to replace the unhealthy mon
-				err = c.failoverMon(mon.Name)
-				if err != nil {
-					logger.Errorf("failed to failover mon %s. %+v", mon.Name, err)
-				}
-			}
-			// only deal with one unhealthy mon per health check
-			return nil
+			c.failMon(len(status.MonMap.Mons), mon.Name)
+			// only deal with one unhealthy mon per health check but deal with
+			//"not found in ceph mon map" too so failover is faster
+			failoverHappening = true
+			continue
```

Comment:
what if we still returned here instead of tracking the `failoverHappening`? Doesn't it just mean that it might take another health check to clean up the monsNotFound?

Reply:
Yes, replaced it by the previous `return`.